### PR TITLE
Feat/back 1441 rfq dynamic taker

### PR DESCRIPTION
**Pull request by mwamedacen**
resolves BACK-1441

### CODE REVIEW

File: src/dex/generic-rfq/fetch-mm-api.ts [hunk 1]
```diff
@@ -31,6 +31,10 @@ import { validateAndCast } from '../../lib/validators';
 
 const network = 1;
 
+const AgustusAddress = {
+  1: '0xdef171fe48cf0115b1d80b88dc8eab59176fee57',
```
Comment (mwamedacen) [inline src/dex/generic-rfq/fetch-mm-api.ts:35]: file used for testing purposes only

File: src/dex/generic-rfq/rate-fetcher.ts [hunk 2]
```diff
@@ -425,6 +427,7 @@ export class RateFetcher {
         this.dexHelper.config.data.network,
```
Comment (KanievskyiDanylo) [inline src/dex/generic-rfq/rate-fetcher.ts:427]: I think it makes sense to add `takerAddress: addressSchema.required(),` to `firmRateResponseValidator.order` validator

File: src/dex/generic-rfq/fetch-mm-api.ts (hunk 1 above)
Comment (KanievskyiDanylo) [reply src/dex/generic-rfq/fetch-mm-api.ts:35]: Typo issue, not a big deal though

File: src/dex/generic-rfq/rate-fetcher.ts (hunk 2 above)
Comment (mwamedacen) [reply src/dex/generic-rfq/rate-fetcher.ts:427]: Actually it was already there but I missed [this validation rule](https://github.com/paraswap/paraswap-dex-lib/blob/master/src/dex/generic-rfq/validators.ts#L171) which enforced augustus addresss. For now it's ok but we'll have to relax it for the future, doing it now

**Pull request merged**